osbs/api.py

```diff
@@ -314,25 +362,30 @@ def _create_build_config_and_build(self, build_request):
             # Reset name change that may have occurred during
             # update above, since renaming is not supported.
             existing_bc['metadata']['name'] = build_config_name
-            logger.debug('build config for %s already exists, updating...', build_config_name)
+            logger.debug('build config for %s already exists, updating...',
+                         build_config_name)
             self.os.update_build_config(build_config_name, json.dumps(existing_bc))
 
+        if image_stream:
+            changed_ist = self.ensure_image_stream_tag(image_stream,
+                                                       image_stream_tag_name,
+                                                       scheduled=True)
+            logger.debug('Changed parent ImageStreamTag? %s', changed_ist)
+
+        if not existing_bc:
```

Comment:
I see, so `ensure_image_stream_tag` has to be run before we create a new BC - or after existing is updated, right?

Lets add a comment about that one, as without knowing this its easy to 'optimize' this